api/v1alpha1/api.go

```diff
@@ -144,3 +150,120 @@ const (
 	// This can be used to describe the routing behavior in HTTPRoute referenced by LLMRoute.
 	LLMModelHeaderKey = "x-envoy-ai-gateway-llm-model"
 )
+
+// BackendSecurityPolicyType specifies the type of auth mechanism used to access a backend.
+type BackendSecurityPolicyType string
+
+const (
+	BackendSecurityPolicyTypeAPIKey BackendSecurityPolicyType = "APIKey"
+	BackendSecurityPolicyTypeAWSIAM BackendSecurityPolicyType = "AWS_IAM"
+)
+
+// +kubebuilder:object:root=true
+
+// BackendSecurityPolicy specifies configuration for authentication and authorization rules on the traffic
+// exiting the gateway to the backend.
+type BackendSecurityPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              BackendSecurityPolicySpec `json:"spec,omitempty"`
+}
+
+// BackendSecurityPolicySpec specifies authentication rules on access the provider from the Gateway.
+type BackendSecurityPolicySpec struct {
+	// Type specifies the auth mechanism used to access the provider. Currently, only "APIKey", AND "AWS_IAM" are supported.
+	//
+	// +kubebuilder:validation:Enum=APIKey;AWS_IAM
+	Type BackendSecurityPolicyType `json:"type"`
+
+	// APIKey is a mechanism to access a backend(s). The API key will be injected into the Authorization header.
+	//
+	// +optional
+	APIKey *AuthenticationAPIKey `json:"apiKey,omitempty"`
+
+	// CloudProviderCredentials is a mechanism to access a backend(s). Cloud provider specific logic will be applied.
+	//
+	// +optional
+	CloudProviderCredentials *AuthenticationCloudProviderCredentials `json:"cloudProviderCredentials,omitempty"`
```

Comment:
so i wonder what kind of logic will make cloud providers special compared to others (currently only apiKey though). how do you envision this additional indirect layer will be used? in other words, what kind of logic will be shared among cloud providers

Reply:
- ApiKey is simple authentication mechanism distributed to user and user is responsible to rotate the keys. 
- Cloud authentication is more sophisticated and support issuing temporary tokens like AWS STS, Google STS to access cloud resources. You do not have to distribute the token and major clouds also support OIDC federation to allow managing your user identity in external system.
see https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp.html#:~:text=You%20can%20use%20the%20AWS,has%20permissions%20to%20do%20so.

Reply:
yeah i get that it's complex, but my question is why this another layer is needed and what kind of logic will be shared among cloud providers? what becomes difficult to achieve when you have `*AWSCredentials` directly here without another layer of `AuthenticationCloudProviderCredentials`. How does having `AuthenticationCloudProviderCredentials` helps implement various cloud providers vs having them in here

Reply:
This is mainly a yaml UX, the additional layer for grouping is there for readability, if we list the individual types then it is not intuitive for user to see the classification. The cloud provider security implementations are different but they share the same high level concepts.